streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -212,6 +218,24 @@
                                         CommonClientConfigs.METRICS_NUM_SAMPLES_DOC);
     }
 
+    // this is the list of configs for underlying clients
+    // that streams prefer different default values
+    private static final Map<String, Object> PRODUCER_OVERRIDES;
+    static
+    {
+        PRODUCER_OVERRIDES = new HashMap<>();
+        PRODUCER_OVERRIDES.put(ProducerConfig.LINGER_MS_CONFIG, "100");
+    }
+
+    private static final Map<String, Object> CONSUMER_OVERRIDES;
```

Comment:
For cases such as this, I think it's good practice to have a temp variable where the map is populated and then wrap it in a Collections.unmodifiableMap while assigning it to the static field. That way we're sure the map is not modified accidentally.